main/cmds.go

```diff
@@ -72,36 +72,44 @@ func install(ctx *log.Context, h HandlerEnvironment, seqNum int) (string, error)
 	// If it is (2) attempt to migrate to mrseq.  Find the latest settings
 	// file, and set the sequence to it.
 	if _, err := os.Stat(mostRecentSequence); os.IsNotExist(err) {
-		migrateToMostRecentSequence(ctx, seqNum)
+		migrateToMostRecentSequence(ctx, h, seqNum)
 	}
 
 	ctx.Log("event", "created data dir", "path", dataDir)
 	ctx.Log("event", "installed")
 	return "", nil
 }
 
-func migrateToMostRecentSequence(ctx *log.Context, seqNum int) {
-	// The configSequenceNumber (seqNum) env variable is used for mrseq if mrseq not found
-	// based on logic in main.go the seqNum should always be populated
-	ctx.Log("event", "recreate mrseq", "message", fmt.Sprintf("cannot find %v", mostRecentSequence))
-	fout, err := os.Create(mostRecentSequence)
+func migrateToMostRecentSequence(ctx *log.Context, h HandlerEnvironment, seqNum int) {
+	// The status folder is used instead of the settings because the settings file is written
+	// by the agent before install is called.  As a result, the extension cannot determine if this
```

Comment:
why is settings file being written by the guest agent before install is called, a problem? is update command called before install? so you have to rely on #.status to know the most recent seq num processed by the previous version of the extn.?

Reply:
Yes exactly, the goal is to get the last run seqnum in the case where the mrseq file is not present. The bug I was originally trying to fix was that the status gets written during update so when we get the seqnum it's the most recent one and not the previous one. One way I was thinking of fixing this is saving the previous seqnum during update scenarios, but open to any ideas you have!